src/components/structures/MatrixChat.js

```diff
@@ -635,23 +657,35 @@ module.exports = React.createClass({
                 action: 'start_post_registration',
             });
         } else if (screen.indexOf('room/') == 0) {
-            var roomString = screen.split('/')[1];
+            var roomString = screen.substring(5);
+            var eventId;
+
+            // extract event id, if one is given
+            var idx = roomString.indexOf('/');
+            if (idx >= 0) {
+                eventId = roomString.substring(idx+1);
+                roomString = roomString.substring(0, idx);
```

Comment:
ok, I see what you mean: we can leave room for future expansion by ignoring path segments we don't understand. My inclination is to have `/room/{roomId}/{eventId}/anything` rather than `/room/{roomId}/anything/{eventId}`, though.